### ISSUE TITLE
[pyFAI] Error observed when using pyFAI-calib2

### ISSUE THREAD

**Issue by picca** [silx-kit/pyFAI#1281]
Hello, I am using the current dev of pyFAI and I get this kind of errors.

```
picca@diffabs6:~/notebooks-soleil/cirpad$ emacs Soleil_Diffabs_Calibration_Cirpad.org
Final residual: 1.891020571821856e-06 (after 1 iterations)
Final residual: 2.0654419112680463e-09 (after 1 iterations)
Final residual: 3.7160486130402986e-09 (after 1 iterations)
ERROR:pyFAI-calib2.UNCAUGHT:float() argument must be a string or a number, not 'NoneType'
Traceback (most recent call last):
  File "/home/experiences/instrumentation/picca/pyFAI/build/lib.linux-x86_64-3.7/pyFAI/gui/tasks/GeometryTask.py", line 969, in __geometryCustomed
    calibration.fromGeometryModel(geometry, resetResidual=True)
  File "/home/experiences/instrumentation/picca/pyFAI/build/lib.linux-x86_64-3.7/pyFAI/gui/helper/RingCalibration.py", line 425, in fromGeometryModel
    self.__geoRef.poni2 = model.poni2().value()
  File "/home/experiences/instrumentation/picca/pyFAI/build/lib.linux-x86_64-3.7/pyFAI/gui/helper/RingCalibration.py", line 74, in __setattr__
    return object.__setattr__(self.__geoRef, name, value)
  File "/home/experiences/instrumentation/picca/pyFAI/build/lib.linux-x86_64-3.7/pyFAI/geometry.py", line 2092, in set_poni2
    self._poni2 = float(value)
TypeError: float() argument must be a string or a number, not 'NoneType'
Final residual: 4.337900633767923e-09 (after 1 iterations)
ERROR:pyFAI-calib2.UNCAUGHT:'<' not supported between instances of 'GeometryRefinementContext' and 'GeometryRefinementContext'
Traceback (most recent call last):
  File "/home/experiences/instrumentation/picca/pyFAI/build/lib.linux-x86_64-3.7/pyFAI/gui/tasks/GeometryTask.py", line 864, in __initGeometry
    self.__initGeometryFromPeaks()
  File "/home/experiences/instrumentation/picca/pyFAI/build/lib.linux-x86_64-3.7/pyFAI/gui/tasks/GeometryTask.py", line 830, in __initGeometryFromPeaks
    calibration.init(peaks, "massif", initialGeometry, constraints)
  File "/home/experiences/instrumentation/picca/pyFAI/build/lib.linux-x86_64-3.7/pyFAI/gui/helper/RingCalibration.py", line 273, in init
    self.__init(peaks, method, geometry, constraintsModel=constraintsModel)
  File "/home/experiences/instrumentation/picca/pyFAI/build/lib.linux-x86_64-3.7/pyFAI/gui/helper/RingCalibration.py", line 257, in __init
    scores.sort()
TypeError: '<' not supported between instances of 'GeometryRefinementContext' and 'GeometryRefinementContext'
```

**Comment by vallsv** [comment]
The first exception is maybe a border case due to a fit calibration
How can we reproduce it?

The second one is interesting, as the sort is proceeded on a tuple, and,  `GeometryRefinementContext` is the last element and was not supposed to be compared. This border case is easy to fix.

**Comment by vallsv** [comment]
Here is one fix, if you want to try #1282

**Comment by vallsv** [comment]
The 1st exception, its maybe reproducible by empty the poni2 text input in the GUI.

I think the better thing to do is to allow `Geometry.set_poni2` to set `poni2` to `None`.
But another problem could occurred later on.
It have to be tested.
